src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/BOM/MaterialForm/MaterialForm.tsx

```diff
@@ -113,10 +114,13 @@ const MaterialForm: React.FC<MaterialFormProps> = ({ submitText, onSubmit, defau
 
   const { assemblies } = wbsElement;
 
-  if (materialTypesIsError) return <ErrorPage message={materialTypesError?.message} />;
-  if (unitsIsError) return <ErrorPage message={unitsError?.message} />;
-  if (manufacturersIsError) return <ErrorPage message={manufacturersError?.message} />;
-  if (
+  const user = useCurrentUser();
```

Comment:
if this component should return `null` for guest users, let's put its condition first before returning any ErrorPage or LoadingIndicator components. 

Then, use auth object from useAuth() function to check if the current user is logged in or not (refer to useUpdateUserRole function in users.hooks.ts to see how to check if user is logged in or not). 

Then, return ErrorPage if either existing ...IsError checker is true or the user is not logged in.